lib/index.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const request = require('request-promise-native');
+const cache = require('./cache');
+
+const defaultInt = 1000;
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        const url = option.url;
+        const interval = parseInt(option.interval, 10) || defaultInt;
+        const currTime = Date.now();
+        const futureTime = currTime + interval;
+        const savedTime = cache.get(url);
+
+        // Is this the first time or has the interval expired
+        if (isNaN(savedTime) || savedTime < currTime) {
+          cache.set(url, futureTime);
+          memo.push(request(url).promise());
+        }
+        return memo;
+
+      }, []);
+
+      // Process all pending requests
+      return Promise.all(requests)
+        .then(resolved => {
+          // resolved is an array
+          for (var i = 0; i < resolved.length; i++) {
+            let response;
+            try {
+              // if response is JSON
+              response = JSON.parse(resolved[i]);
+            } catch (e) {
+              response = resolved[i];
+            }
+
+            if (response) {
+              // find the status code
+              const status = response.statusCode || response.status;
+              // If there is a status code
+              // but it's not a 200, throw an Error
+              if (status) {
+                if (parseInt(status, 10) !== 200) {
```

Comment:
I'd check for a 2xx rather than a strict 200. It would be reasonable for a healthcheck to respond with 204 for example. Arguably a 304 *might* be legit too.

FWIW, this is how hof-model detects a "success" - https://github.com/UKHomeOfficeForms/hof-model/blob/master/lib/model.js#L179